File: Februrary 2025/23.cpp
class Solution {
  public:
    // Function to find the next greater element for each element of the array.
    vector<int> nextLargerElement(vector<int>& arr) {
        int n = arr.size();
        vector<int> result(n, -1);
        stack<int> s;
        
        for (int i = 0; i < n; ++i) {
            while (!s.empty() && arr[s.top()] < arr[i]) {
                result[s.top()] = arr[i];
                s.pop();
            }
            s.push(i);
        }
        
        return result;
    }
};

File: Februrary 2025/23.py
class Solution:
    # Function to find the next greater element for each element of the array.
    def nextLargerElement(self, arr):
        n = len(arr)
        result = [-1] * n
        stack = []

        for i in range(n):
            while stack and arr[stack[-1]] < arr[i]:
                result[stack.pop()] = arr[i]
            stack.append(i)
        
        return result
